pkg/util/node/node.go

```diff
@@ -139,23 +139,35 @@ func GetNodeIP(client clientset.Interface, hostname string) net.IP {
 
 // GetZoneKey is a helper function that builds a string identifier that is unique per failure-zone;
 // it returns empty-string for no zone.
+// Since there are currently two separate zone keys:
+//   * "failure-domain.beta.kubernetes.io/zone"
+//   * "topology.kubernetes.io/zone"
+// GetZoneKey will first check failure-domain.beta.kubernetes.io/zone and if not exists, will then check
+// topology.kubernetes.io/zone
 func GetZoneKey(node *v1.Node) string {
 	labels := node.Labels
 	if labels == nil {
 		return ""
 	}
 
-	region, _ := labels[v1.LabelZoneRegion]
-	failureDomain, _ := labels[v1.LabelZoneFailureDomain]
+	zone, ok := labels[v1.LabelZoneFailureDomain]
```

Comment:
Add a TODO to prefer the stable key in 1.18ish timeframe

Reply:
done